packages/firestore/src/api/field_value.ts

```diff
@@ -107,11 +107,12 @@ export class ArrayUnionFieldValueImpl extends SerializableFieldValue {
 
   _toFieldTransform(context: ParseContext): FieldTransform {
     // Although array transforms are used with writes, the actual elements
-    // being uniomed or removed are not considered writes since they cannot
+    // being unioned or removed are not considered writes since they cannot
     // contain any FieldValue sentinels, etc.
     const parseContext = new ParseContext(
```

Comment:
It might be time to add a helper method to ParseContext (or a tree-shakeable function in this file) that simplifies the creation of an argument parser. If `arrayElement` remains configurable, this code would replace line 112-122, 145-155 and 173-181.

Optional.